<h1>Activity 2 - UK Road Accident Data Analytics</h1>
<hr>

In [1]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

In [2]:
road_accidents = pd.read_csv('datasets\\uk_accident_data.csv')
#road_accidents

In [3]:
road_accidents.columns

Index(['Index', 'Accident_Severity', 'Accident Date', 'Latitude',
       'Light_Conditions', 'District Area', 'Longitude',
       'Number_of_Casualties', 'Number_of_Vehicles', 'Road_Surface_Conditions',
       'Road_Type', 'Urban_or_Rural_Area', 'Weather_Conditions',
       'Vehicle_Type'],
      dtype='object')

<h1>Extrating Date Information</h1>

In [4]:
road_accidents['Accident Date'] = pd.to_datetime(road_accidents['Accident Date'], dayfirst = True, errors = 'coerce')

In [5]:
road_accidents['Year'] = road_accidents['Accident Date'].dt.year
road_accidents['Month'] = road_accidents['Accident Date'].dt.month
road_accidents['Day'] = road_accidents['Accident Date'].dt.day
road_accidents['Week'] = road_accidents['Accident Date'].dt.dayofweek

road_accidents

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious    2019-06-05  51.506187   
1       200701BS65737           Serious    2019-07-02  51.495029   
2       200701BS66127           Serious    2019-08-26  51.517715   
3       200701BS66128           Serious    2019-08-16  51.495478   
4       200701BS66837            Slight    2019-09-03  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    2022-02-18  57.374005   
660675  201091NM01881            Slight    2022-02-21  57.232273   
660676  201091NM01935            Slight    2022-02-23  57.585044   
660677  201091NM01964           Serious    2022-02-23  57.214898   
660678  201091NM02142           Serious    2022-02-28  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  Year  Month  Day  Week  
0                                        Car  2019      6    5     2  
1                                        Car  2019      7    2     1  
2                      Taxi/Private hire car  2019      8   26     0  
3       Bus or coach (17 or more pass seats)  2019      8   16     4  
4                              Other vehicle  2019      9    3     1  
...                                      ...   ...    ...  ...   ...  
660674                 

<h1>Fill Null Values</h1>

In [6]:
road_accidents.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                      25
Light_Conditions               0
District Area                  0
Longitude                     26
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions      726
Road_Type                   4520
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
Year                           0
Month                          0
Day                            0
Week                           0
dtype: int64

In [7]:
road_accidents['Latitude'] = road_accidents['Latitude'].fillna(road_accidents['Latitude'].mean())
road_accidents['Longitude'] = road_accidents['Longitude'].fillna(road_accidents['Longitude'].mean())
road_accidents['Road_Surface_Conditions'] = road_accidents['Road_Surface_Conditions'].fillna(road_accidents['Road_Surface_Conditions'].mode()[0])
road_accidents['Road_Type'] = road_accidents['Road_Type'].fillna(road_accidents['Road_Type'].mode()[0])
road_accidents['Urban_or_Rural_Area'] = road_accidents['Urban_or_Rural_Area'].fillna(road_accidents['Urban_or_Rural_Area'].mode()[0])
road_accidents['Weather_Conditions'] = road_accidents['Weather_Conditions'].fillna(road_accidents['Weather_Conditions'].mode()[0])

road_accidents.isnull().sum()

Index                      0
Accident_Severity          0
Accident Date              0
Latitude                   0
Light_Conditions           0
District Area              0
Longitude                  0
Number_of_Casualties       0
Number_of_Vehicles         0
Road_Surface_Conditions    0
Road_Type                  0
Urban_or_Rural_Area        0
Weather_Conditions         0
Vehicle_Type               0
Year                       0
Month                      0
Day                        0
Week                       0
dtype: int64

<h1>Change Data Types</h1>

In [8]:
road_accidents['Accident_Severity'] = road_accidents['Accident_Severity'].astype('category')
road_accidents['Light_Conditions'] = road_accidents['Light_Conditions'].astype('category')
road_accidents['District Area'] = road_accidents['District Area'].astype('category')
road_accidents['Road_Surface_Conditions'] = road_accidents['Road_Surface_Conditions'].astype('category')
road_accidents['Road_Type'] = road_accidents['Road_Type'].astype('category')
road_accidents['Urban_or_Rural_Area'] = road_accidents['Urban_or_Rural_Area'].astype('category')
road_accidents['Weather_Conditions'] = road_accidents['Weather_Conditions'].astype('category')
road_accidents['Vehicle_Type'] = road_accidents['Vehicle_Type'].astype('category')

road_accidents.dtypes

Index                              object
Accident_Severity                category
Accident Date              datetime64[ns]
Latitude                          float64
Light_Conditions                 category
District Area                    category
Longitude                         float64
Number_of_Casualties                int64
Number_of_Vehicles                  int64
Road_Surface_Conditions          category
Road_Type                        category
Urban_or_Rural_Area              category
Weather_Conditions               category
Vehicle_Type                     category
Year                                int32
Month                               int32
Day                                 int32
Week                                int32
dtype: object

<hr>
<h1>Questions:</h1>
<h3>1. How many accidents are considered slight, serious or fatal? </h3>

In [9]:
road_accidents['Accident_Severity'].value_counts()

Accident_Severity
Slight     563801
Serious     88217
Fatal        8661
Name: count, dtype: int64

<h1>Insight #1</h1>
<h3>There are 563,801 slight accidents, 88,217 serious accidents and 8,661 fatal accidents.</h3>
<hr>
<h3>2. Does accidents happen more on urban or rural area?
</h3>

In [10]:
road_accidents['Urban_or_Rural_Area'].mode()

0    Urban
Name: Urban_or_Rural_Area, dtype: category
Categories (3, object): ['Rural', 'Unallocated', 'Urban']

<h1>Insight #2</h1>
<h3>Urban areas experiences more accidents than rural areas.</h3>
<hr>
<h3>3. What is the highest number of casualties in the road accidents?
</h3>

In [11]:
road_accidents.describe().T

count                           mean  \
Accident Date           660679  2020-11-30 08:30:32.761749760   
Latitude              660679.0                      52.553866   
Longitude             660679.0                       -1.43121   
Number_of_Casualties  660679.0                        1.35704   
Number_of_Vehicles    660679.0                       1.831255   
Year                  660679.0                     2020.40909   
Month                 660679.0                       6.607965   
Day                   660679.0                       15.58135   
Week                  660679.0                       3.111195   

                                      min                  25%  \
Accident Date         2019-01-01 00:00:00  2019-11-27 00:00:00   
Latitude                         49.91443            51.490696   
Longitude                       -7.516225            -2.332191   
Number_of_Casualties                  1.0                  1.0   
Number_of_Vehicles                    1.0                  1.0   
Year                               2019.0               2019.0   
Month                                 1.0                  4.0   
Day                                   1.0                  8.0   
Week                                  0.0                  1.0   

                                      50%                  75%  \
Accident Date         2020-11-13 00:00:00  2021-11-17 00:00:00   
Latitude                        52.315755            53.453444   
Longitude                       -1.411761            -0.232901   
Number_of_Casualties                  1.0                  1.0   
Number_of_Vehicles                    2.0                  2.0   
Year                               2020.0               2021.0   
Month                                 7.0                 10.0   
Day                                  16.0                 23.0   
Week                                  3.0                  5.0   

                                      max       std  
Accident Date         2022-12-31 00:00:00       NaN  
Latitude                        60.757544  1.406896  
Longitude                         1.76201  1.383302  
Number_of_Casualties                 68.0  0.824847  
Number_of_Vehicles                   32.0  0.715269  
Year                               2022.0  1.109152  
Month                                12.0  3.420701  
Day                                  31.0  8.764863  
Week                                  6.0  1.927284

In [12]:
road_accidents['Number_of_Casualties'].max()

68

<h1>Insight #3</h1>
<h3>The highest number of casualties in a single road accident is 68.</h3>
<hr>
<h3>4. What type of vehicle experience the most accidents?
</h3>

In [13]:
road_accidents['Vehicle_Type'].value_counts()

Vehicle_Type
Car                                      497992
Van / Goods 3.5 tonnes mgw or under       34160
Bus or coach (17 or more pass seats)      25878
Motorcycle over 500cc                     25657
Goods 7.5 tonnes mgw and over             17307
Motorcycle 125cc and under                15269
Taxi/Private hire car                     13294
Motorcycle over 125cc and up to 500cc      7656
Motorcycle 50cc and under                  7603
Goods over 3.5t. and under 7.5t            6096
Other vehicle                              5637
Minibus (8 - 16 passenger seats)           1976
Agricultural vehicle                       1947
Pedal cycle                                 197
Data missing or out of range                  6
Ridden horse                                  4
Name: count, dtype: int64

<h1>Insight #4</h1>
<h3>Cars experiences the most accidents with 497,992 accidents.</h3>
<hr>
<h3>5. What is the average number of casualties and vehicles in the road accidents?</h3>

In [14]:
road_accidents['Number_of_Casualties'].mean()

1.357040257068864

In [15]:
road_accidents['Number_of_Vehicles'].mean()

1.8312554205597575

<h1>Insight #5</h1>
<h3>The average number of casualties is 1.35 and the average number of vehicles is 1.83.</h3>
<hr>
<h3>6. How many accidents have more than 1 vehicle involved?
</h3>

In [16]:
(road_accidents['Number_of_Vehicles'] > 1).value_counts()

Number_of_Vehicles
True     459892
False    200787
Name: count, dtype: int64

<h1>Insight #6</h1>
<h3>There are 459,892 accidents that have more than 1 vehicle involved in it.</h3>
<hr>
<h3>7. Which district have the lowest amount of road accidents?</h3>

In [17]:
road_accidents['District Area'].value_counts()

District Area
Birmingham            13491
Leeds                  8898
Manchester             6720
Bradford               6212
Sheffield              5710
                      ...  
Berwick-upon-Tweed      153
Teesdale                142
Shetland Islands        133
Orkney Islands          117
Clackmannanshire         91
Name: count, Length: 422, dtype: int64

<h1>Insight #7</h1>
<h3>The district area of Clackmannanshire have the lowest with only having 91 road accidents.</h3>
<hr>
<h3>8. What date have the highest amount of accidents in a single day?</h3>

In [18]:
road_accidents['Accident Date'].value_counts()

Accident Date
2019-11-30    704
2019-01-31    697
2019-07-13    692
2021-11-13    692
2019-08-14    688
             ... 
2022-12-30    171
2019-12-25    157
2022-12-25    145
2022-01-10    123
2020-12-25    118
Name: count, Length: 1461, dtype: int64

<h1>Insight #8</h1>
<h3>In 30-11-2019, the most accidents happened with about 704 being recorded that day.</h3>
<hr>
<h3>9. Base on the latitude, what location does most accidents happen?</h3>

In [19]:
road_accidents['Latitude'].value_counts()

Latitude
52.458798    50
52.949719    45
53.083165    33
52.967634    31
52.938860    29
             ..
53.657705     1
53.628677     1
53.643256     1
53.629944     1
57.575210     1
Name: count, Length: 511619, dtype: int64

<h1>Insight #9</h1>
<h3>The area with the 52.458798 for its latitude have the highest amount of accidents with 50.</h3>
<hr>
<h3>10.  Which area has more fatal accidents, urban or rural?</h3>

In [20]:
more_fatal = road_accidents.groupby(['Urban_or_Rural_Area', 'Accident_Severity']).size()
more_fatal

Urban_or_Rural_Area  Accident_Severity
Rural                Fatal                  5601
                     Serious               37312
                     Slight               196077
Unallocated          Fatal                     0
                     Serious                   1
                     Slight                   10
Urban                Fatal                  3060
                     Serious               50904
                     Slight               367714
dtype: int64

<h1>Insight #10</h1>
<h3>Rural areas experiences more fatal accidents with it having 2,541 more than the urban areas.</h3>
<hr>
<h3>11. What is the total number of casualties from the dataset?</h3>

In [21]:
road_accidents["Number_of_Casualties"].sum()

896568

<h1>Insight #11</h1>
<h3>There are a total of 896,568 casualties from all the accidents in the dataset.</h3>
<hr>
<h3>12. Which area has more total number of casualties, Urban or Rural?</h3>

In [22]:
road_accidents.groupby("Urban_or_Rural_Area")["Number_of_Casualties"].sum()

Urban_or_Rural_Area
Rural          353515
Unallocated        13
Urban          543040
Name: Number_of_Casualties, dtype: int64

<h1>Insight #12</h1>
<h3>Urban areas have more number of casualties with the combined number of 543,040.</h3>
<hr>
<h3>13. Which accident severity have the highest avereage number of vehicles?</h3>

In [23]:
road_accidents.groupby("Accident_Severity")["Number_of_Vehicles"].mean()

Accident_Severity
Fatal      1.786976
Serious    1.678327
Slight     1.855864
Name: Number_of_Vehicles, dtype: float64

<h1>Insight #13</h1>
<h3>Slight accidents have the highest average number of vehicles with an average of 1.85.</h3>
<hr>
<h3>14. Which road type have the lowest average number of casualties?</h3>

In [24]:
road_accidents.groupby("Road_Type")["Number_of_Casualties"].mean()

Road_Type
Dual carriageway      1.477279
One way street        1.192713
Roundabout            1.274891
Single carriageway    1.343788
Slip road             1.423661
Name: Number_of_Casualties, dtype: float64

<h1>Insight #14</h1>
<h3>One way streets have the lowest average number of casualties with an average of 1.19.</h3>
<hr>
<h3>15. Which weather condition causes the most fatal accidents?</h3>

In [25]:
fatal_weather = road_accidents.groupby(['Weather_Conditions', 'Accident_Severity']).size()
fatal_weather

Weather_Conditions     Accident_Severity
Fine + high winds      Fatal                   175
                       Serious                1245
                       Slight                 7134
Fine no high winds     Fatal                  7207
                       Serious               73285
                       Slight               454521
Fog or mist            Fatal                    82
                       Serious                 483
                       Slight                 2963
Other                  Fatal                   165
                       Serious                1801
                       Slight                15184
Raining + high winds   Fatal                   145
                       Serious                1261
                       Slight                 8209
Raining no high winds  Fatal                   848
                       Serious                9468
                       Slight                69380
Snowing + high winds   Fatal             

<h1>Insight #15</h1>
<h3>The weather condition 'Fine + high winds' causes the most fatal accidents.</h3>
<hr>
<h3>16. Which light condition causes the least amount of serious accidents?</h3>

In [26]:
light = road_accidents.groupby(['Light_Conditions', 'Accident_Severity']).size()
light

Light_Conditions             Accident_Severity
Darkness - lighting unknown  Fatal                    68
                             Serious                 794
                             Slight                 5622
Darkness - lights lit        Fatal                  1860
                             Serious               19130
                             Slight               108345
Darkness - lights unlit      Fatal                    45
                             Serious                 360
                             Slight                 2138
Darkness - no lighting       Fatal                  1612
                             Serious                7174
                             Slight                28651
Daylight                     Fatal                  5076
                             Serious               60759
                             Slight               419045
dtype: int64

<h1>Insight #16</h1>
<h3>The light condition 'Darkness - light unlit' causes the least amount of serious accidents.</h3>
<hr>
<h3>17. Which combination of road type and light conditions causes the least amount of accidents?</h3>

In [27]:
road_light = road_accidents.groupby(['Road_Type', 'Light_Conditions']).size()
road_light

Road_Type           Light_Conditions           
Dual carriageway    Darkness - lighting unknown       905
                    Darkness - lights lit           20242
                    Darkness - lights unlit           465
                    Darkness - no lighting           7460
                    Daylight                        70352
One way street      Darkness - lighting unknown       124
                    Darkness - lights lit            3310
                    Darkness - lights unlit            58
                    Darkness - no lighting            103
                    Daylight                         9964
Roundabout          Darkness - lighting unknown       470
                    Darkness - lights lit           10528
                    Darkness - lights unlit           189
                    Darkness - no lighting            224
                    Daylight                        32581
Single carriageway  Darkness - lighting unknown      4899
                    Dark

<h1>Insight #17</h1>
<h3>Slippy road with darkness - lights unlit have the least amount of accidents with only having 34 accidents.</h3>
<hr>
<h3>18. Which combination of weather and road surface causes the most accidents?</h3>

In [28]:
surface_weather = road_accidents.groupby(['Road_Surface_Conditions', 'Weather_Conditions']).size()
surface_weather

Road_Surface_Conditions  Weather_Conditions   
Dry                      Fine + high winds          5026
                         Fine no high winds       439733
                         Fog or mist                 473
                         Other                      2634
                         Raining + high winds         72
                         Raining no high winds       551
                         Snowing + high winds          4
                         Snowing no high winds        54
Flood over 3cm. deep     Fine + high winds            15
                         Fine no high winds          132
                         Fog or mist                   3
                         Other                        19
                         Raining + high winds        291
                         Raining no high winds       548
                         Snowing + high winds          3
                         Snowing no high winds         6
Frost or ice             Fine + high wind

<h1>Insight #18</h1>
<h3>The combination of dry road and fine no high winds weather causes the most accidents.</h3>
<hr>
<h3>19. Which combination of weather and road surface causes the most accidents?</h3>

In [29]:
road_type_surface = road_accidents.groupby(['Road_Type', 'Road_Surface_Conditions']).size()
road_type_surface

Road_Type           Road_Surface_Conditions
Dual carriageway    Dry                         66272
                    Flood over 3cm. deep          302
                    Frost or ice                 2408
                    Snow                          909
                    Wet or damp                 29533
One way street      Dry                         10100
                    Flood over 3cm. deep            5
                    Frost or ice                  183
                    Snow                           76
                    Wet or damp                  3195
Roundabout          Dry                         30784
                    Flood over 3cm. deep           22
                    Frost or ice                  745
                    Snow                          232
                    Wet or damp                 12209
Single carriageway  Dry                        336671
                    Flood over 3cm. deep          679
                    Frost or ice      

<h1>Insight #19</h1>
<h3>A single carriegeway road with dry surface causes the most accidents with 336,671 accidentS.</h3>
<hr>
<h3>20. Is there a correlation between the number of casualties and the vehicles involved?</h3>

In [30]:
casualty_vehicle_corr = road_accidents['Number_of_Casualties'].corr(road_accidents['Number_of_Vehicles'])
casualty_vehicle_corr

0.2288888612692756

<h1>Insight #20</h1>
<h3>There is a no correlation between the number of casualties and the number of vehicle in the road accidents.</h3>
<hr>

In [34]:
# What year have the highest record of fatal accidents?

year = road_accidents.groupby(['Year','Accident_Severity']).size()
year

Year  Accident_Severity
2019  Fatal                  2714
      Serious               24322
      Slight               155079
2020  Fatal                  2341
      Serious               23121
      Slight               145129
2021  Fatal                  2057
      Serious               21997
      Slight               139500
2022  Fatal                  1549
      Serious               18777
      Slight               124093
dtype: int64

In [39]:
# What month have the least amount of combined casualties?

road_accidents.groupby("Month")["Number_of_Casualties"].sum()

Month
1     71038
2     66513
3     73076
4     70620
5     76796
6     76296
7     78490
8     75280
9     76053
10    80302
11    81453
12    70651
Name: Number_of_Casualties, dtype: int64

In [37]:
year = road_accidents.groupby(['Weather_Conditions','Month']).size()
year.unstack()

Month                     1      2      3      4      5      6      7      8   \
Weather_Conditions                                                              
Fine + high winds       1816    694   1448    276    463    225    315    298   
Fine no high winds     36754  38759  44754  46870  48732  49465  47689  45251   
Fog or mist              434    615    209    140     54    102     36     52   
Other                   2948   1996    952    553    634    553    713    664   
Raining + high winds    1990    501   1055    118    449    292    429    390   
Raining no high winds   7307   4964   5218   3596   6003   5828   8247   7245   
Snowing + high winds     188    216    162     23      3      0      1      2   
Snowing no high winds   1435   1746    288    168     14     16     15     11   

Month                     9      10     11     12  
Weather_Conditions                                 
Fine + high winds        426    579   1388    626  
Fine no high winds     48536  49132  42244  36827  
Fog or mist              128    278    577    903  
Other                    663   1188   2173   4113  
Raining + high winds     520    710   2217    944  
Raining no high winds   6167   7615  11197   6309  
Snowing + high winds       2     11     65    212  
Snowing no high winds     13     67    563   1902

In [49]:
year = road_accidents.groupby(['Road_Surface_Conditions','Week']).size()
year.unstack()

Week                         0      1      2      3      4      5      6
Road_Surface_Conditions                                                 
Dry                      49433  63687  65630  68026  67314  73155  61302
Flood over 3cm. deep       161    171    127    106    104    182    166
Frost or ice              2259   2878   2978   3281   2197   2477   2447
Snow                       740    991    816    849   1007    856    631
Wet or damp              20087  26823  30007  27249  27278  30508  24756

In [52]:
year = road_accidents.groupby(['Month','District Area']).size()
year.unstack()

District Area  Aberdeen City  Aberdeenshire  Adur  Allerdale  Alnwick  \
Month                                                                   
1                         92            149    58         89       23   
2                        120            133    41         85       25   
3                        114            171    53         82       14   
4                         90            134    58         86       17   
5                        123            161    58        101       16   
6                        111            187    58         89       18   
7                        112            170    51         91       16   
8                        118            170    48        114       25   
9                        103            150    43         93       18   
10                       121            165    53        107       23   
11                       113            171    59         90       18   
12                       106            169    39        101       19   

District Area  Amber Valley  Angus  Argyll and Bute  Arun  Ashfield  ...  \
Month                                                                ...   
1                       116     64               53   110       131  ...   
2                       100     89               58   100        98  ...   
3                       116     55               58   104       108  ...   
4                        98     53               63   121       116  ...   
5                        97     70               73   125       105  ...   
6                       120     71               73   113       125  ...   
7                       127     58               78   136       140  ...   
8                       119     51               95    90       102  ...   
9                       123     73               89   125       104  ...   
10                      124     61               77   137       121  ...   
11                      106     87               56   102       130  ...   
12                      101     64               63   113       115  ...   

District Area  Wokingham  Wolverhampton  Worcester  Worthing  Wrexham  \
Month                                                                   
1                    102            222         61        64      122   
2                    101            196         72        66      106   
3                    110            193         73        90      106   
4                    103            204         78        81      103   
5                    119            219         77        84      122   
6                    117            202         60        87       99   
7                    101            211         62        88       96   
8                     85            183         76        97       91   
9                    124            201         85        91      113   
10                   108            206         71        88      120   
11                   115            242         67        92      125   
12                   125            187         76        80       99   

District Area  Wychavon  Wycombe  Wyre  Wyre Forest  York  
Month                                                      
1                    93      129    89           77   191  
2                   106      120    93           73   130  
3                   102      152   114           74   164  
4                   123      132   115          101   144  
5                   139      144   109           82   149  
6                   113      138    96           89   153  
7                   111      164    87           75   139  
8                   111      128   114           71   147  
9                   105      146    84           85   139  
10                  127      174   110          101   188  
11                  120      164   121           76   201  
12                  111      138   106           65   152  

[12 rows x 422 columns]